p2p/queue.go

```diff
@@ -33,9 +33,9 @@ type fifoQueue struct {
 
 var _ queue = (*fifoQueue)(nil)
 
-func newFIFOQueue() *fifoQueue {
+func newFIFOQueue(size int) *fifoQueue {
```

Comment:
/cc @erikgrinaker -- in order to get this to play nicely with the p2p test network stack and the consensus reactor, we need these channels to be buffered. In the router, of course, we maintain them to be unbuffered (i.e. 0).

Not sure if you had thoughts on this as far as APIs and what should be buffered, how much, and where.